### PR TITLE
AmirHossein-Javascript1-Week1-Homework

### PR DESCRIPTION
**Pull request by amirhshad**

### CODE REVIEW

File: Week2/homework/js-exercises/drinks-ex5.js [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+  count++;
+  if (count ==2) {
+    pointer++;
+    count = 0;
+  };
+  if (pointer >= drinkTypes.length) {
+    pointer = 0;
+  };
+};
+console.log(drinkTray);
```
Comment (n4cr) [inline Week2/homework/js-exercises/drinks-ex5.js]: You just copy pasted my code. Please follow the requirements and log the right statement at the end

File: Week2/homework/js-exercises/readinglist-ex4.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+    {
+        title: "Deep Work",
+        author: "Kal Newport",
+        alreadyRead: false,
+    }
+]
+
+
+
+books.forEach((element, index, array) => {
```
Comment (n4cr) [inline Week2/homework/js-exercises/readinglist-ex4.js:22]: I like that you use declarative loop

File: Week2/homework/js-exercises/readinglist-ex4.js [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+    console.log(element.title + " by " + element.author);
+    if (element.alreadyRead === true) {
```
Comment (n4cr) [inline Week2/homework/js-exercises/readinglist-ex4.js]: For a boolean you don't need `===`. simply say `if (element.alreadyRead)`

File: Week2/homework/js-exercises/recipe-ex3.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+// for (let property in recipe) {
+//     console.log(`${property}: ${recipe[property]}`);
+// };
+
+//Method3:
+let property = Object.keys(recipe);
+let value = Object.values(recipe);
+
+for (let i = 0; i < Object.keys(recipe).length; i++) {
+
```
Comment (n4cr) [inline Week2/homework/js-exercises/recipe-ex3.js:23]: Object.values is not necessarily sorted. Therefore it is possible that you get [2, 'pizza', [..]] instead. Which means you can possibly get Meal name: 2. You should be careful how you use it.

File: Week2/homework/js-exercises/recipe-ex3.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+"use strict"
+let recipe = {
+    mealName : "Mirzaghasemi",
+    serves: 2,
+    ingredients : ["2 eggplants", "2 eggs", "1 garlic"]
+};
+//Method1:
+// for (let property in recipe) {
+//     console.log(property + ": " + recipe[property]);
```
Comment (n4cr) [inline Week2/homework/js-exercises/recipe-ex3.js:10]: the ordering of lines in the output might be different in other browsers. You should be careful if ordering matters. or you need a different solution.

By the way you solutions are hacky. your output should be:

Meal name: ...
Serves: ..
Ingredients: ...

yours are just keys of the dictionary. please adjust it

**Review by n4cr — CHANGES_REQUESTED** [review]
Looks good. I've reuqested some changes though

File: Week2/homework/js-exercises/drinks-ex5.js (hunk 1 above)
Comment (amirhshad) [reply Week2/homework/js-exercises/drinks-ex5.js]: Just resolved.

File: Week2/homework/js-exercises/readinglist-ex4.js (hunk 3 above)
Comment (amirhshad) [reply Week2/homework/js-exercises/readinglist-ex4.js]: resolved

File: Week2/homework/js-exercises/recipe-ex3.js (hunk 4 above)
Comment (amirhshad) [reply Week2/homework/js-exercises/recipe-ex3.js:23]: I think I need more explanation! Let's do it in today's meeting

**Pull request merged**